src/Components/DisplayToDos.js

```diff
@@ -0,0 +1,128 @@
+import { useState, useEffect } from "react";
+
+import firebase from "../firebase";
+import { getDatabase, ref, onValue, update, remove } from 'firebase/database';
+
+import CloseButton from "./CloseButton";
+
+const DisplayToDos = ({ handleClose }) => {
+    const [toDos, setToDos] = useState([]);
+
+    useEffect(() => {
+        const database = getDatabase(firebase);
+        const dbRef = ref(database);
+        onValue(dbRef, (res) => {
+            const responseArr = [];
+            const response = res.val();
+            //change object response from firebase into an array
+            for (let key in response) {
+                responseArr.push({ key: key, data: response[key] });
+            }
+            setToDos(responseArr);
+        })
+    }, []);
+
+
+    const handleVisitedButton = (toDo) => {
+        const database = getDatabase(firebase);
+        const dbRef = ref(database, `/${toDo.key}`);
+        return update(dbRef, {isDone: true})
+    };
+
+    const handleDeleteButton = (toDo) => {
+        const database = getDatabase(firebase);
+        const dbRef = ref(database, `/${toDo.key}`);
+        remove(dbRef);
+    };
+
+    const handleUnvisitedButton = (toDo) => {
+        const database = getDatabase(firebase);
+        const dbRef = ref(database, `/${toDo.key}`);
+        return update(dbRef, {isDone: false})
+    };
+
```

Comment:
Your field notes area works seamlessly and i found it intuitive! I love that you figured out how add/remove options and your event handlers are cleanly written & easy to follow.

Reply:
🙏